R/project-element.R

```diff
@@ -0,0 +1,124 @@
+#'             project_title = "my project title", 
+#'             award_information = add_funding(funder_name = "Bank",
+#'                                             funder_identifier = "Funder 1",
+#'                                             award_number = "000",
+#'                                             award_title = "Money up for grabs",
+#'                                             award_url = "awardforme.com"),
+#'             project_personnel = add_personnel(parent_element = parent_element,
+#'                                               first_name = "Smithy",
+#'                                               last_name = "Smith",
+#'                                               email = "myemail@mail.gov",
+#'                                               role = "Manager",
+#'                                               organization = "US GOV"))
+#'                                               
+#' add_project(parent_element = parent_element, 
+#'             project_title = "my project title", 
+#'             award_information = list(funderName = "Bank",
+#'                                      funderIdentifier = "Funder 1",
+#'                                      awardNumber = "000",
+#'                                      title = "Money up for grabs",
+#'                                      awardUrl = "awardforme.com"),
+#'             project_personnel = list(individualName = list(givenName = "Smithy",
+#'                                                            surName = "Smith"),
+#'                                      electronicMailAddress = "myemail@mail.gov",
+#'                                      role = "Manager",
+#'                                      organizationName = "US GOV"))
+
+add_project <- function(parent_element, project_title, 
+                        award_information, project_personnel) {
+  
+  required_arguments <- c("project_title", "award_infomration", "project_personnel")
+  
+  missing_argument_index <- which(c(missing(project_title), missing(award_information),
+                                    missing(project_personnel)))
+  
+  if (length(missing_argument_index) > 0) {
+    project_error <- required_arguments[missing_argument_index][1]
+    project_error_message <- switch(project_error, 
+                                    project_title = "Please provide the project title",
+                                    award_infomration = "Please provide the award information.",
+                                    project_personnel = "Please provide the project personnel")
+    
+    if (missing(project_title) | missing(award_information) |
+        missing(project_personnel)) {
+      stop(project_error_message, call. = FALSE)
+    }
+  }
+  required_information <- c("funderName", "title", "individualName", "organization",
+                            "role")
+  
+  null_required_information <- which(c(is.null(award_information$funderName), 
```

Comment:
you could likely combine the missing and null error handling, would be nice to revisit when i am back